index.js

```diff
@@ -194,9 +201,13 @@ yargs
   .command('cut [level]', 'cuts a release branch from origin/master', (yargs) => {
     yargs
       .positional('level', {
-        choices: ['major', 'minor'],
+        choices: ['major', 'minor', 'premajor', 'preminor', 'prerelease'],
         default: 'minor',
         describe: 'The level of the release'
+      })
+      .option('preid', {
```

Comment:
Would you mind updating the documentation and/or with an example of canonical preid usage? Would this be used in conjunction with premajor/preminor/prerelease level? I'm not quite sure how those are intended interact.

I'm also not sure a pre designation warrants a separate release branch name. I would think that a release that is tagged v1.0.0-pre for example would be on the v1.0 release branch instead of it's own.

Reply:
> Would you mind updating the documentation and/or with an example of canonical preid usage?

Yep, I'll do that. 

> Would this be used in conjunction with premajor/preminor/prerelease level

You do not need to pass the `preid` option. e.g.: `semver.inc('1.2.3', 'premajor'); // Outputs 2.0.0-0`. Using `preid` is up to the user to determine if they want more flexibility there, e.g. if the user wanted to distinguish between alpha and beta for a prerelease patch increment. Passing the `preid` option when doing non-prerelease increments is benign, i.e. `semver.inc('1.2.3', 'major', 'beta')` will result in `2.0.0`. At least, that's what [the documentation ](https://github.com/npm/node-semver#prerelease-identifiers) indicates, and how it worked when I tested it.

> I'm also not sure a pre designation warrants a separate release branch name.

Yeah, I don't feel strongly about this. My thought was to make it explicit so that there wouldn't be confusion when looking at a branch called `v1.0` and it really being `1.0.0-rc.5` but I suppose it is a little opinionated. I can revert that change.

Reply:
One question/concern I do have though is that currently running `git cactus cut` always applies the version increment commits *to the master branch*. This is probably not good if you are doing a prerelease, yeah? So maybe I remove the portion that cuts a prerelease named branch, and instead, have the version commits only happen on the cut branch?

Reply:
`git cactus cut` clones the repository's origin master branch in a temporary location so it is entirely independent of what your situation is locally with your working repo.

`git cactus tag` tags HEAD where you are in your working repo.

This may be confusing but the reasoning is as follows:

 - When Friday's branch cut rolls around, you should be able to cut no matter what the state of your tree is. You are logically cutting master from origin, not from your local dev tree.
 - When you are tagging a specific follow up, you should be able to specify the exact point to tag using your local tree state (today's functionality) or referencing by a particular sha (future improvement).

Reply:
Ah, so `cut` will always be from `origin/master`? If that's the case, I'm not sure this is mergeable since the workflow for a prerelease would likely not want to increment master, but just the branch that's cut. Maybe I'm misunderstanding the cactus branching workflow, but my guess would be that if you had a prerelease 1.0.0, but your current stable is 0.9.0, you'd probably want master to stay 0.9.0. The way I'm understanding this, `origin master` will always be incremented with `cut`?